pandas/tests/indexes/test_category.py

```diff
@@ -507,6 +507,22 @@ def test_identical(self):
         self.assertTrue(ci1.identical(ci1.copy()))
         self.assertFalse(ci1.identical(ci2))
 
+    def test_ensure_copied_data(self):
+        # Check the "copy" argument of each Index.__new__ is honoured
+        # GH12309
+        # Must be tested separately from other indexes because
+        # self.value is not an ndarray
+        for index in self.indices.values():
+            result = CategoricalIndex(index.values, copy=True)
+            tm.assert_index_equal(index, result)
+            if index.values.base is not None:
+                self.assertIsNot(index.values.base, self.result.base)
+
+            result = CategoricalIndex(index.values, copy=False)
+            bases = [i.values if i.values.base is None else i.values.base
```

Comment:
Will add the tests... but I don't get the "allow ``None``" part: to what argument?

By the way, at a second glance the fact that ``assert_numpy_array_equal`` now ensures objects are ndarrays doesn't make much sense, at least as long as the docstring says:

    left : np.ndarray or iterable
[...]

    check_dtype: bool, default True
        check dtype if both a and b are np.ndarray

@sinhrks what do you think?